datalake/src/main/java/com/sequenceiq/datalake/service/sdx/ProvisionerService.java

```diff
@@ -245,14 +240,18 @@ private AwsNetworkV4Parameters convertToAwsNetwork(EnvironmentNetworkResponse so
         return response;
     }
 
-    private void setupAuthentication(StackV4Request stackRequest) {
+    private void setupAuthentication(DetailedEnvironmentResponse environment, StackV4Request stackRequest) {
         if (stackRequest.getAuthentication() == null) {
             StackAuthenticationV4Request stackAuthenticationV4Request = new StackAuthenticationV4Request();
-            stackAuthenticationV4Request.setPublicKey(dummySshKey);
+            stackAuthenticationV4Request.setPublicKey(getSshKey(environment));
             stackRequest.setAuthentication(stackAuthenticationV4Request);
         }
     }
 
+    private String getSshKey(DetailedEnvironmentResponse environment) {
+        return environment.getAuthentication().getPublicKey();
```

Comment:
Can we put a null check here and throw exception if there is no authentication in environment? With this code it can throw NPE, but would be better if we throw more precise exception if there is no authentication part in env.